daemon/list_test.go

```diff
@@ -79,7 +81,53 @@ func containerListContainsName(containers []*containertypes.Summary, name string
 	return false
 }
 
-func TestListInvalidFilter(t *testing.T) {
+func TestContainerList(t *testing.T) {
+	db, err := container.NewViewDB()
+	assert.NilError(t, err)
+	d := &Daemon{
+		containersReplica: db,
+	}
+
+	// test list with no containers
+	t.Run("no containers", func(t *testing.T) {
+		containerList, err := d.Containers(context.Background(), &containertypes.ListOptions{All: true})
+		assert.NilError(t, err)
+		assert.Assert(t, is.Len(containerList, 0))
+	})
+
+	// test list with one or more containers
+	t.Run("more containers", func(t *testing.T) {
+		for num := range []int{1, 2, 4, 8, 16, 32, 64, 100} {
```

Comment:
We could run each case in a separate sub-test:
```suggestion
		for num := range []int{1, 2, 4, 8, 16, 32, 64, 100} {
			t.Run(strconv.Itoa(num), func(t *testing.T) {
```

otherwise, if one of them fails, we won't be able to immediately know which case failed

Reply:
Good idea, done.